internal/server/routes.go

```diff
@@ -186,6 +189,21 @@ func add[Req, Res any](a *API, r *gin.RouterGroup, route route[Req, Res]) {
 	bindRoute(a, r, route.method, route.path, wrappedHandler)
 }
 
+func trimWhitespace(req interface{}) {
+	v := reflect.ValueOf(req)
+	for v.Type().Kind() == reflect.Pointer && !v.IsNil() {
+		v = v.Elem()
+	}
+	if v.Kind() == reflect.Struct {
+		for i := 0; i < v.NumField(); i++ {
+			f := v.Field(i)
+			if f.Kind() == reflect.String {
```

Comment:
minor: I think this will also match types like:

```go
type Custom string
```

If we want to avoid those I think this might work

```go
strType := reflect.TypeOf("")
...
if f.Type() == strType {
```

Not a blocker if there's a reason to prefer the comparison to kind.